packages/kpi-domains-readmodel-checker/src/utils.ts

```diff
@@ -0,0 +1,106 @@
+import { diff } from "json-diff";
+import isEqual from "lodash.isequal";
+import { Logger } from "pagopa-interop-commons";
+import { WithMetadata } from "pagopa-interop-models";
+
+function getIdentificationKey<T extends { id: string } | { kid: string }>(
+  obj: T
+): string {
+  return "id" in obj ? obj.id : obj.kid;
+}
+
+export function zipDataById<T extends { id: string } | { kid: string }>(
+  dataA: Array<WithMetadata<T>>,
+  dataB: Array<WithMetadata<T>>
+): Array<[WithMetadata<T> | undefined, WithMetadata<T> | undefined]> {
+  const allIds = new Set(
+    [...dataA, ...dataB].map((d) => getIdentificationKey(d.data))
+  );
+  return Array.from(allIds).map((id) => [
+    dataA.find((d) => getIdentificationKey(d.data) === id),
+    dataB.find((d) => getIdentificationKey(d.data) === id),
+  ]);
+}
```

Comment:
This function doesn't seem to be used, should be removed?

```suggestion
export function zipDataById<T extends { id: string } | { kid: string }>(
  dataA: Array<WithMetadata<T>>,
  dataB: Array<WithMetadata<T>>
): Array<[WithMetadata<T> | undefined, WithMetadata<T> | undefined]> {
  const allIds = new Set(
    [...dataA, ...dataB].map((d) => getIdentificationKey(d.data))
  );
  return Array.from(allIds).map((id) => [
    dataA.find((d) => getIdentificationKey(d.data) === id),
    dataB.find((d) => getIdentificationKey(d.data) === id),
  ]);
}
```